### DISCUSSION TITLE
[BUG] Cannot change date/location of external media items

### DISCUSSION THREAD

**Discussion by andriykrefer** [immich-app/immich#6397] [Q&A]
### The bug

- Selected an item (synced from a external media folder) and tried to change date or location. Got a warning saying it cannot change its metadata
- Done the same test on a uploaded items and it worked.

I understand that external media items are read only and the original files are not touched. But as far I understood, Immich reads the file metadata and stores it in the database, so it should be possible to change the date/location because it should update the database entry only. Also, it is stated in the documentation that external media items should behave exactly like uploaded items.

### The OS that Immich Server is running on

Cant tell, running on pikapods.com

### Version of Immich Server

v1.92.0

### Version of Immich Mobile App

v1.92.0 build 117

### Platform with the issue

- [X] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Cant tell, running on pikapods.com (Fully managed by pikapods.com)
```

### Your .env content

```Shell
Cant tell, running on pikapods.com (Fully managed by pikapods.com)
```

### Reproduction steps

```bash
(Tested in the web app and android app)

- Selected an item (synced from a external media folder) and tried to change date or location. Got a warning saying it cannot change its metadata
- Done the same test on a uploaded items and it worked.
```

**Comment by schuhbacca** [comment]
It's not just a database change, it writes a sidecar metadata file for the image. This doesn't seem like a bug more a feature request since it's working as intended.

**Reply by jrasm91** [reply]
As of right now we don't support the exit writing features for external libraries.

**Reply by andriykrefer** [reply]
Sorry, I don't get the sidecar and exif stuff. Wouldn't it be enough only write to database, in the case of external media?

**Reply by schuhbacca** [reply]
A sidecar file is a file that lives alongside the media file and provides metadata. Immich creates these and writes the location and date data if you edit it though the UI. 

You can't only write to the database. If you lose the DB you lose all that data, but if the sidecar file exists the metadata can just be re-extracted. The DB is not the source of truth, the media and sidecar is.

**Reply by jrasm91** [reply]
The bigger issue is that running extract metadata again in the asset would overwrite the database changes. They would essentially get lost and that is quite confusing and frustrating.

**Reply by andriykrefer** [reply]
I understand, but don't know what is worse, because it is very inconvenient to change wrong dates in external media currently. I would have to find the photo files, change the metadata and perform a re-scan. After the re-scan, the updated photos will be recognized as new items and lose all album and favorite assignments (I suppose, haven't tested it), and the old items will be shown as offline duplicates.
I guess the recommended next step would be to open a feature request to allow the metadata change of external libraries?

**Reply by schuhbacca** [reply]
They shouldn't be recognized as new items. The hash for external libraries is made from the file path. So if you update the file metadata it shouldn't be recognized as a new file, and once it's re-scanned the edits should show up. But yes the eventual goal will allow editing and deleting I believe. If you can't find the feature request already feel free to create one!
